### PR TITLE
PR for Network Policy lab material

### PR DESCRIPTION
**Pull request by filiy**

### CODE REVIEW

**Review by NickCorcoran — COMMENTED** [review]
Great stuff Phil.  I like it so far.  I do have a few comments:
Line 160 - allow from same namespace
- Are you planning a step where we only allow comms between 2 specific components?  I'm thinking like a database that doesn't need to be exposed, but needs to talk to web or api server.


Line 184 Allow from OpenShift Router
- Does this allow ingress to hit just a route?  If yes, that's good.  If it can hit any pod, not so good.

**Comment by filiy** [comment]
Thanks for the feedback @NickCorcoran ! 

Ok, after the line 160 ## Allow from the same namespace section I'll add another section to only allow specfic pods to talk to each other.

The line 184 ## Allow from OpenShift Router section is allowing traffic from the ingress router pods to any pod in the project. External traffic will only come in to the cluster on 80 & 443 or just 443 to the ingress router pods then directed to project pods based on the routes. Is that fine? This rule is the generic allow routes to hit pods rule.

**Comment by NickCorcoran** [comment]
> Thanks for the feedback @NickCorcoran !
> 
> Ok, after the line 160 ## Allow from the same namespace section I'll add another section to only allow specfic pods to talk to each other.
> 
> The line 184 ## Allow from OpenShift Router section is allowing traffic from the ingress router pods to any pod in the project. External traffic will only come in to the cluster on 80 & 443 or just 443 to the ingress router pods then directed to project pods based on the routes. Is that fine? This rule is the generic allow routes to hit pods rule.

That is fine :)  Thanks.

**Comment by filiy** [comment]
@NickCorcoran This lab is all complete now and ready for a final review. Once it's all reviewed and updated or approved we can merge.

**Comment by NickCorcoran** [comment]
This looks good Phil.

File: openshift-201/network-policy.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
+# Network Policies and ACS
+
+By default, all pods in a project are accessible from other pods and network endpoints. To isolate one or more pods in a project, you can create NetworkPolicy objects in that project to indicate the allowed incoming connections. Project administrators can create and delete NetworkPolicy objects within their own project.
```
Comment (mtspn) [inline openshift-201/network-policy.md:3]:
```suggestion
Without a network policy in place, all pods in a project are accessible from other pods and network endpoints. To isolate one or more pods in a project, you can create NetworkPolicy objects in that project to indicate the allowed incoming connections. Project administrators can create and delete NetworkPolicy objects within their own project. A 'deny by default' policy is automatically created and active in all projects on the platform. Later in this lab we look at how to build a list of allowed list onto the default of denying network connections. 
```

**Pull request merged**